### ISSUE TITLE
关于Tensor索引的建议和文档的建议

### ISSUE THREAD

**Issue by huochf** [PaddlePaddle/Paddle#29768]
建议添加整形数组或布尔数组索引，就如同numpy和pytorch那样！！！！

例如在pytorch中，可以写出很简短的代码：
`target_classes_o = torch.cat([t["labels"][J] for t, (_, J) in zip(targets, indices)]) # [batch_size, num_objects]`
而翻译成paddle，却
```
target_classes_o = [t["labels"].numpy()[J.numpy()] for t, (_, J) in zip(targets, indices)]
target_classes_o = [dg.to_variable(t) for t in target_classes_o]
target_classes_o = L.concat(target_classes_o) # [bs, num_object]
```

建议加速paddle官网文档的更新，我看咱源码函数头部里面的注释本身就很清楚，详细了！可以先导出，先用着，现在我都得在github的paddle代码中找API。
